api/v1/common_types.go

```diff
@@ -6,6 +6,26 @@ type Image struct {
 	Git *GitSource `json:"git,omitempty"`
 	// Name of container image (example: "docker.io/your-username/your-image").
 	Name string `json:"name,omitempty"`
+	// Upload is a signal that a local tar of the directory should be uploaded to be built as an image.
+	Upload *UploadSpec `json:"upload,omitempty"`
+}
+
+type UploadSpec struct {
```

Comment:
`type ImageUpload` ... We shouldnt suffix all things with Spec.

Reply:
here I'm looking for a way to distinguish `UploadSpec` from `UploadStatus`. Do you have a good way to do that using `ImageUpload` as the spec representation?